packages/experiments-realm/address.gts

```diff
@@ -4,25 +4,89 @@ import {
   Component,
   FieldDef,
 } from 'https://cardstack.com/base/card-api';
-import StringCard from 'https://cardstack.com/base/string';
+import StringField from 'https://cardstack.com/base/string';
+import { CountryField } from './country';
+
+function getAddressRows(
+  addressLine1: string | undefined,
+  addressLine2: string | undefined,
+  city: string | undefined,
+  state: string | undefined,
+  postalCode: string | undefined,
+  countryCode: string | undefined,
+  poBoxNumber: string | undefined,
+) {
+  const rows: string[][] = [[], [], []];
+
+  if (addressLine1) {
+    rows[0].push(addressLine1);
+  }
+  if (addressLine2) {
+    rows[0].push(addressLine2);
+  }
+
+  const cityStatePostal = [city, state, postalCode].filter(Boolean).join(', ');
+
+  if (cityStatePostal) {
+    rows[1].push(cityStatePostal);
+  }
+
+  if (countryCode) {
+    rows[2].push(countryCode);
+  }
+
+  if (poBoxNumber) {
+    rows[3].push(`PO Box ${poBoxNumber}`);
+  }
+  return rows;
+}
 
 export class Address extends FieldDef {
   static displayName = 'Address';
-  @field streetAddress = contains(StringCard); // required
-  @field city = contains(StringCard); // required
-  @field region = contains(StringCard);
-  @field postalCode = contains(StringCard);
-  @field poBoxNumber = contains(StringCard);
-  @field country = contains(StringCard); // required // dropdown
+  @field addressLine1 = contains(StringField);
+  @field addressLine2 = contains(StringField);
+  @field city = contains(StringField);
+  @field state = contains(StringField);
+  @field postalCode = contains(StringField);
+  @field country = contains(CountryField);
+  @field poBoxNumber = contains(StringField);
+  @field fullAddress = contains(StringField, {
+    computeVia: function (this: Address) {
+      let rows = getAddressRows(
+        this.addressLine1,
+        this.addressLine2,
+        this.city,
+        this.state,
+        this.postalCode,
+        this.country?.name,
+        this.poBoxNumber,
+      );
+      return rows.filter((r) => r.length > 0).join(', ');
```

Comment:
@tintinthong if we agree to use the suggestion above, then we can just

```
 return rows.join(',\n');
```

Added `\n` so it can push another row of address to next line

Reply:
No I specifically used this to be one single line so it can be used with a maps api